From c259ccdf4b1ed257142333c4f540eb1c7616bb51 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Tue, 25 Jun 2024 17:05:02 +0200
Subject: [PATCH 0969/1695] core: arm: fix calloc() against GCC 14
 -Wcalloc-transposed-args

Fix calloc() calls against GCC 14 directive -Wcalloc-transposed-args that
makes GCC to complain with an warning/error trace message like:

warning: 'calloc' sizes specified with 'sizeof' in the earlier argument and not in the later argument [-Wcalloc-transposed-args]

No functional change.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 1db4ff98b4057dab5dd94b2a06e8e28711de59d4 core: arm: fix calloc() against GCC 14 -Wcalloc-transposed-args]
Change-Id: I21d00194a60409891aa845db06c44e6f31d72bff
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
Acked-by: Gatien Chevallier <gatien.chevallier@foss.st.com>
Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/390272
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 core/arch/arm/kernel/thread_spmc.c | 2 +-
 core/arch/arm/mm/sp_mem.c          | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/core/arch/arm/kernel/thread_spmc.c b/core/arch/arm/kernel/thread_spmc.c
index 477a43b5b..3c2c687aa 100644
--- a/core/arch/arm/kernel/thread_spmc.c
+++ b/core/arch/arm/kernel/thread_spmc.c
@@ -988,7 +988,7 @@ static int add_mem_share(struct ffa_mem_transaction_x *mem_trans,
 		return FFA_NO_MEMORY;
 
 	if (flen != blen) {
-		struct mem_frag_state *s = calloc(sizeof(*s), 1);
+		struct mem_frag_state *s = calloc(1, sizeof(*s));
 
 		if (!s) {
 			rc = FFA_NO_MEMORY;
diff --git a/core/arch/arm/mm/sp_mem.c b/core/arch/arm/mm/sp_mem.c
index bdba7309b..aa9134581 100644
--- a/core/arch/arm/mm/sp_mem.c
+++ b/core/arch/arm/mm/sp_mem.c
@@ -224,7 +224,7 @@ struct sp_mem *sp_mem_new(void)
 	uint32_t exceptions = 0;
 	int i = 0;
 
-	smem = calloc(sizeof(*smem), 1);
+	smem = calloc(1, sizeof(*smem));
 	if (!smem)
 		return NULL;
 
-- 
2.47.2

